File: python/Candy.py
class Solution:
    # @param ratings, a list of integer
    # @return an integer
    def candy(self, ratings):
        amount=[]
        r=[]
        ans=[]
        step=1
        for i in range(len(ratings)):
            amount.append(1)
            r.append(1)
            ans.append(1)
        for i in range(1,len(ratings)):
            if ratings[i]>ratings[i-1]:
                amount[i]=amount[i-1]+1
        #print('left',amount)
        for i in range(len(ratings)-1,0,-1):
            if ratings[i-1]>ratings[i]:
                r[i-1]=r[i]+1
        #print('right',r)
        for i in range(len(ratings)):
            #print(i,amount[i],r[i])
            ans[i]=max(amount[i],r[i])
        #print('ans',ans)
        return sum(ans)
        
s=Solution()
array=[2,1]
#print(array)
print(s.candy(array))

File: Array/Candy.java
import java.util.*;
public class Candy {
    public int candy(int[] ratings) {
        int [] amount = new int [ratings.length];
        Arrays.fill(amount,1);
        int [] left = amount.clone();
        int [] right= amount.clone();
		for(int i =1;i<ratings.length;i++){
            if(ratings[i]>ratings[i-1]) left[i]=left[i-1]+1;
        }
        for(int i=ratings.length-2;i>=0;i--){
            if(ratings[i]>ratings[i+1]) right[i]=right[i+1]+1;
        }		
        int sum=0;
        for(int i=0;i<amount.length;i++){
            amount[i]=Math.max(left[i],right[i]);
            sum+=amount[i];
        }
        return sum;
    }
	public static void main(String args[]){
		Candy c= new Candy();
		int [] rating={1,2,3,1,2,3,1,2,2,1,2,1,1,4};
		System.out.println(c.candy(rating));
	}
}